container_service_extension/configure_cse.py

```diff
@@ -121,47 +145,45 @@ def check_cse_installation(config, msg_update_callback=utils.NullPrinter()):
                 msg_update_callback.error(msg)
                 SERVER_CLI_LOGGER.error(msg)
                 err_msgs.append(msg)
-        except Exception:  # TODO() replace raw exception with specific
-            msg = f"Could not connect to AMQP exchange '{amqp['exchange']}'"
-            msg_update_callback.error(msg)
-            SERVER_CLI_LOGGER.error(msg)
-            err_msgs.append(msg)
-        finally:
-            if connection is not None:
-                connection.close()
+            finally:
+                if connection is not None:
+                    connection.close()
 
-        # check that CSE is registered to vCD correctly
-        ext = api_extension.APIExtension(client)
-        try:
-            cse_info = ext.get_extension(server_constants.CSE_SERVICE_NAME,
-                                         namespace=server_constants.CSE_SERVICE_NAMESPACE) # noqa: E501
-            rkey_matches = cse_info['routingKey'] == amqp['routing_key']
-            exchange_matches = cse_info['exchange'] == amqp['exchange']
-            if not rkey_matches or not exchange_matches:
-                msg = "CSE is registered as an extension, but the extension " \
-                      "settings on vCD are not the same as config settings."
-                if not rkey_matches:
-                    msg += f"\nvCD-CSE routing key: {cse_info['routingKey']}" \
-                           f"\nCSE config routing key: {amqp['routing_key']}"
-                if not exchange_matches:
-                    msg += f"\nvCD-CSE exchange: {cse_info['exchange']}" \
-                           f"\nCSE config exchange: {amqp['exchange']}"
-                msg_update_callback.info(msg)
-                SERVER_CLI_LOGGER.info(msg)
+            # check that CSE is registered to vCD correctly
+            ext = api_extension.APIExtension(client)
+            try:
+                cse_info = ext.get_extension(server_constants.CSE_SERVICE_NAME,
+                                             namespace=server_constants.CSE_SERVICE_NAMESPACE) # noqa: E501
+                rkey_matches = cse_info['routingKey'] == amqp['routing_key']
```

Comment:
Pardon my ignorance, but something doesn;t look right here.
We added a check above in the same function to see if we need to configure MQTT or AMQP.
Now at this stage are we guaranteed to have AMQP configs?
i.e. amqp['routing_key'], amqp['exchange'] etc?

Reply:
This is a helpful point. I think this touches on your earlier comment we tagged @rocknes in. We are currently assuming that amqp is the default, and mqtt is optional to use instead if the user specifies.

 In get_validated_config in container_service_extension/config_validator.py, this code makes sure that the config will either have an "mqtt" or "amqp" section, depending on the the use_mqtt_protocol helper in utils. Therefore, at this point in the code, if MQTT is not used, there is an AMQP section. This is a roundabout reasoning, but we have the guarantee from the config validator that we will have either the amqp or mqtt sections; otherwise, we would have had an error earlier.